models/experimental/functional_distilbert/tests/test_perf_distilbert.py

```diff
@@ -0,0 +1,158 @@
+        question,
+        context,
+        max_length=seq_len,
+        padding="max_length",
+        truncation=True,
+        return_attention_mask=True,
+        return_tensors="pt",
+    )
+
+    profiler.start(f"preprocessing_parameter")
+    parameters = preprocess_model_parameters(
+        model_name=f"ttnn_{model_name}_optimized",
+        initialize_model=lambda: DistilBertForQuestionAnswering.from_pretrained(model_name, torchscript=False).eval(),
+        custom_preprocessor=ttnn_optimized_distilbert.custom_preprocessor,
+        device=device,
+    )
+    profiler.end(f"preprocessing_parameter")
+
+    with torch.no_grad():
+        profiler.start(cpu_key)
+        torch_out = hugging_face_reference_model(**inputs)
+        profiler.end(cpu_key)
+
+        durations = []
+        for _ in range(2):
+            profiler.start(f"preprocessing_input")
+            position_ids = None
+            input_ids, position_ids, attention_mask = ttnn_optimized_distilbert.preprocess_inputs(
+                inputs["input_ids"],
+                position_ids,
+                inputs["attention_mask"],
+                device=device,
+            )
+            profiler.end(f"preprocessing_input")
+
+            start = time.time()
+            tt_output = ttnn_optimized_distilbert.distilbert_for_question_answering(
+                config,
+                input_ids=input_ids,
+                attention_mask=attention_mask,
+                position_ids=position_ids,
+                parameters=parameters,
+                device=device,
+            )
+            tt_output = ttnn.from_device(tt_output)
+            end = time.time()
+
+            durations.append(end - start)
+            enable_persistent_kernel_cache()
+
+    inference_and_compile_time, inference_time, *_ = durations
+
+    expected_compile_time, expected_inference_time = get_expected_times_qa()
+
+    prep_perf_report(
+        model_name=f"ttnn_{model_name}_optimized",
+        batch_size=batch_size,
+        inference_and_compile_time=inference_and_compile_time,
+        inference_time=inference_time,
+        expected_compile_time=expected_compile_time,
+        expected_inference_time=expected_inference_time,
+        comments="",
+        inference_time_cpu=0.0,
+    )
+
+    logger.info(f"Compile time: {inference_and_compile_time - inference_time}")
+    logger.info(f"Inference time: {inference_time}")
+    logger.info(f"Samples per second: {1 / inference_time * batch_size}")
+
+    assert (
+        inference_time < expected_inference_time
+    ), f"Expected inference time: {expected_inference_time} Actual inference time: {inference_time}"
+    logger.info("Exit SD perf test")
+
+
+@pytest.mark.models_device_performance_bare_metal
+@pytest.mark.parametrize(
+    "batch_size, test, expected_perf",
+    [
+        [8, "384-8-phiyodr/bert-large-finetuned-squad2", 260],
+    ],
+)
+def test_distilbert_perf_device(batch_size, test, expected_perf, reset_seeds):
```

Comment:
Why would you not add this to the device perf suite?

Reply:
You used the `models_device_performance_bare_metal` with the expected perf as a device cycles count. This means you mean to add it to the device perf suite: https://github.com/tenstorrent/tt-metal/actions/workflows/perf-device-models.yaml

However, I do not see this test run in the section of the performance script for device performance.

Reply:
Will add this test to device_perf suite.